frontend/src/Judge/reducers.js

```diff
@@ -50,6 +54,19 @@ const queue = (state = initialQueueState, action) => {
   const { order, viewing } = state
 
   switch (action.type) {
+    case actions.FETCH_SUBMISSIONS:
+      const {
+        submissions,
+        username
+      } = action.payload
+      // Merge in the new submissions and perform a shuffle
+      const submissionIds = submissions.map(submission => submission.id)
+      const mergedOrder = Array.from(new Set([...order, ...submissionIds]))
```

Comment:
Can you explain the purpose of `mergedOrder`? Why do we need to account for `...order`?

Reply:
So.... I'm a bit uncertain as to whether this action is pulling in _all_ of the entries or just some of them. I wrote it assuming the latter, but yeah if it's the former then this isn't necessary.

Reply:
The `Submissions` query takes a required `showId` param, and we call `fetchSubmissions()` when we load a particular show to vote on, so theoretically, the entries should have the same `id`